src/biokbase/narrative/tests/test_appmanager.py

```diff
@@ -887,13 +900,75 @@ def test_bulk_app_no_wsid(self, c):
         c.return_value.send_comm_message = comm_mock
         self.run_app_expect_error(
             comm_mock,
-            lambda: self.am.run_app_bulk(self.bulk_run_good_inputs),
+            lambda: self.am.run_app_bulk(get_bulk_run_good_inputs()),
             "run_app_bulk",
             'Unable to retrieve system variable: "workspace_id"',
         )
 
     ############# End tests for run_app_bulk #############
 
+    def test_reconstitute_shared_params(self):
+        app_info_el = {
+            "shared_params": {
+                "shared_param_key0": "shared_param_val0",
+                "shared_param_key1": "shared_param_val1"
+            },
+            "params": [
+                {
+                    "param_key00": "param_val00",
+                    "param_key01": "param_val01"
+                }, {
+                    "param_key10": "param_val10",
+                    "param_key11": "param_val11"
+                }
+            ],
+            "other_key": "other_val"
+        }
+
+        # Merge shared_params into each params dict
+        self.am._reconstitute_shared_params(app_info_el)
+        self.assertEqual(
+            {
+                "params": [
+                    {
+                        "param_key00": "param_val00",
+                        "param_key01": "param_val01",
+                        "shared_param_key0": "shared_param_val0",
+                        "shared_param_key1": "shared_param_val1"
+                    }, {
+                        "param_key10": "param_val10",
+                        "param_key11": "param_val11",
+                        "shared_param_key0": "shared_param_val0",
+                        "shared_param_key1": "shared_param_val1"
+                    }
+                ],
+                "other_key": "other_val"
+            },
```

Comment:
can you make this into its own variable for clarity/brevity, so that the tests become
```py
self.assertEqual(app_info_el, expected)

# No shared_params means no change
self.am._reconstitute_shared_params(app_info_el)
self.assertEqual(app_info_el, expected)
```

Reply:
Good catch